flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/ChangelogStateBackendHandle.java

```diff
@@ -113,6 +124,19 @@ public long getStateSize() {
                     + nonMaterialized.stream().mapToLong(StateObject::getStateSize).sum();
         }
 
+        @Override
+        public long getIncrementalStateSize() {
+            long incrementalStateSize =
+                    incrementalMaterializeSize == undefinedIncrementalMaterializeSize
+                            ? materialized.stream()
+                                    .mapToLong(StateObject::getIncrementalStateSize)
+                                    .sum()
+                            : incrementalMaterializeSize;
+            incrementalStateSize +=
+                    nonMaterialized.stream().mapToLong(StateObject::getIncrementalStateSize).sum();
+            return incrementalStateSize;
```

Comment:
This method can be **potentially** called for every rendering (though I think currently it is cached on JM).
Maybe we should we compute the value once in constructor?